### PR TITLE
AW-115 - Fix History loading for seeded accounts

### PR DESCRIPTION
**Pull request by centrys-otniel**

### CODE REVIEW

File: src/main/java/org/aion/wallet/ui/components/partials/UnlockAccountDialog.java [hunk 1]
```diff
@@ -94,7 +96,8 @@ public void resetValidation() {
     @Subscribe
     private void handleUnlockStarted(final AccountEvent event) {
         if (AccountEvent.Type.UNLOCKED.equals(event.getType())) {
-            this.account = event.getAccount();
+            this.account = event.getPayload();
+//            unlockPassword.requestFocus();
```
Comment (centrys-cristian-ilca) [inline src/main/java/org/aion/wallet/ui/components/partials/UnlockAccountDialog.java]: remove comment

File: src/main/java/org/aion/wallet/connector/api/ApiBlockchainConnector.java [hunk 2]
```diff
@@ -275,6 +294,75 @@ private void handleSettingsChanged(final SettingsEvent event) {
...
+    }
+
+    private BlockDTO getOldestSafeBlock(final Set<String> addresses, final Consumer<Iterator<String>> nullSafeBlockFilter) {
+        BlockDTO oldestSafeBlock = null;
+        final Iterator<String> addressIterator = addresses.iterator();
+        while (addressIterator.hasNext()) {
+            final String address = addressIterator.next();
+            final BlockDTO lastSafeBlock = getAccountManager().getLastSafeBlock(address);
+            if (lastSafeBlock != null) {
+                if (oldestSafeBlock == null) {
```
Comment (centrys-victor) [inline src/main/java/org/aion/wallet/connector/api/ApiBlockchainConnector.java]: can be refactored into `oldestSafeBlock == null || oldestSafeBlock.getNumber() > lastSafeBlock.getNumber()`

File: src/main/java/org/aion/wallet/ui/components/partials/UnlockAccountDialog.java [hunk 3]
```diff
@@ -42,6 +43,7 @@
     @Override
     public void initialize(final URL location, final ResourceBundle resources) {
         registerEventBusConsumer();
+        Platform.runLater(() -> unlockPassword.requestFocus());
```
Comment (centrys-victor) [inline src/main/java/org/aion/wallet/ui/components/partials/UnlockAccountDialog.java]: Thanks for this one

File: src/main/java/org/aion/wallet/ui/MainWindow.java [hunk 4]
```diff
@@ -40,8 +40,8 @@
     private static final String TITLE = "Aion Wallet";
     private static final String MAIN_WINDOW_FXML = "MainWindow.fxml";
     private static final String AION_LOGO = "components/icons/aion-icon.png";
-
-    private final BlockchainConnector blockchainConnector = BlockchainConnector.getInstance();
+    private static final String LOCAL_DIR = System.getProperty("user.dir");
```
Comment (centrys-victor) [inline src/main/java/org/aion/wallet/ui/MainWindow.java]: Couldn't this be local.storage.dir ?

File: src/main/java/org/aion/wallet/ui/components/partials/UnlockAccountDialog.java (hunk 3 above)
Comment (centrys-otniel) [reply src/main/java/org/aion/wallet/ui/components/partials/UnlockAccountDialog.java]: :+1:

File: src/main/java/org/aion/wallet/ui/MainWindow.java (hunk 4 above)
Comment (centrys-otniel) [reply src/main/java/org/aion/wallet/ui/MainWindow.java]: actually no, beacuse this is where the app is started from, so we need this

File: src/main/java/org/aion/wallet/connector/api/ApiBlockchainConnector.java (hunk 2 above)
Comment (centrys-otniel) [reply src/main/java/org/aion/wallet/connector/api/ApiBlockchainConnector.java]: will do

File: src/main/java/org/aion/wallet/ui/components/partials/UnlockAccountDialog.java (hunk 3 above)
Comment (centrys-otniel) [reply src/main/java/org/aion/wallet/ui/components/partials/UnlockAccountDialog.java]: This is my single most satisfying feature

**Pull request merged**